runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -450,7 +450,7 @@ class MM_ObjectAllocationAPI
 		uintptr_t dataSize = ((uintptr_t)size) * J9ARRAYCLASS_GET_STRIDE(arrayClass);
 		bool validSize = true;
 #if !defined(J9VM_ENV_DATA64)
-		validSize = !sizeCheck || (size < ((uint32_t)J9_MAXIMUM_INDEXABLE_DATA_SIZE / J9ARRAYCLASS_GET_STRIDE(arrayClass)));
+		validSize = !sizeCheck || (0 == J9ARRAYCLASS_GET_STRIDE(arrayClass)) || (size < ((uint32_t)J9_MAXIMUM_INDEXABLE_DATA_SIZE / J9ARRAYCLASS_GET_STRIDE(arrayClass)));
```

Comment:
please create another variable for `J9ARRAYCLASS_GET_STRIDE(arrayClass)` and use it instead of calling macro twice